src/main/java/com/google/collegeplanner/servlets/PlannerServlet.java

```diff
@@ -29,35 +35,221 @@
 /** Servlet that returns the mutli-semester plan for the given courses.*/
 @WebServlet("/api/planner")
 public class PlannerServlet extends HttpServlet {
+  private HashMap<String, Integer> indegree; // Number of prerequisites for a course
+  private HashMap<String, HashSet<String>>
+      nextCourses; // Courses that come after the given course in the graph
+  private HashMap<String, HashSet<String>> corequisites;
+  private HashMap<String, Integer> depths; // depth of a course in the graph
+  private HashMap<String, Integer> credits; // Number of credits for each course
+  private ArrayList<String> courseList;
+  private int totalCredits;
+
   /**
-   * Organizes courses into the given number of semesters
+   * Organizes courses from POST request into a given number of semesters
    */
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    // TODO(ramyabuva): Implement algorithm to split into the number of semesters
     JSONObject body;
+    int numSemesters;
+    JSONArray selectedClasses;
     try {
       body = getBody(request);
-    } catch (ParseException | NullPointerException e) {
+      numSemesters = Integer.parseInt((String) body.get("semesters"));
+      selectedClasses = (JSONArray) body.get("selectedClasses");
+    } catch (Throwable e) {
       respondWithError(
           "Invalid body for POST request.", HttpServletResponse.SC_BAD_REQUEST, response);
       return;
     }
-    String strClasses = (String) body.get("selectedClasses");
-    // Returns the courses in a single semester
-    String[][] courses = {strClasses.split(",")};
+
+    constructGraphs(selectedClasses);
+
+    ArrayList<ArrayList<String>> semesters = getPlan(numSemesters);
+    // If courseList is not empty, scheduling was not possible
+    if (!courseList.isEmpty()) {
+      return;
```

Comment:
looks like you're returning an empty body, instead of an empty json array. Let's make sure you have at least something in the return output.

In addition, I would wrap the output in an Object, so it's more future-proof:

```
{
  semesters: ...
  some_other_future_field: ...
}
```